### DISCUSSION TITLE
Mac M1 and `weasyprint` package

### DISCUSSION THREAD

**Discussion by OttoAndrey** [pyenv/pyenv#2439] [Q&A]
Hello. I have the same issue like here https://github.com/pyenv/pyenv/issues/2339
And I don't know how to fix this.

My setup:
MacOs 12.5.1 (M1)
pyenv 2.3.3
python 3.10.2 (also tried 3.10.6 - same error)

Steps to reproduce inside virtualenv created through pyenv
```
brew install pango
pip install weasyprint
python -c "import weasyprint"
```

and I got `OSError: ctypes.util.find_library() did not manage to locate a library called 'fontconfig'`

**Comment by native-api** [comment]
Did you rather mean https://github.com/pyenv/pyenv/issues/2339 ?

Pyenv doesn't change anything in Ctypes. Whatever the problem is, it's in CPython's logic. So I suggest to either study that to find out how it can be tweaked to your needs and/or consulting with the CPython community about that.

The linked issue claims that Ctypes "only searches predefined paths" which is false (those paths are rather a fallback) and suggests to "fix" it with a patch that patches searching for FFI during Python's build rather than Ctypes' general library search logic and is presumably already present in newer CPython releases.

**Reply by OttoAndrey** [reply]
Yeah, you are right. I made a typo when inserted link. Edited.

I tried new version 3.10.6 - same problem :(

**Comment by OttoAndrey** [comment]
I installed python through brew and used venv. Now weasyprint works fine. But if I do it through pyenv - error

**Comment by teramike** [comment]
What worked for me:

Install your python version required through homebrew e.g. `brew install python@3.8`
Create your venv with that homebrew installed python: `/opt/homebrew/bin/python3.8 -m venv venv` 

Install `weasyprint` with pip install, if you already installed all other required external libraries with homebrew it  should work.
